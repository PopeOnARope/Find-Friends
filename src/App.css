@font-face {
  font-family: bebas;
  src: url(./assets/BEBAS.ttf);
}
