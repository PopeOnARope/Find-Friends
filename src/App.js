import React, { Component } from "react";
import { connect } from "react-redux";
import { toggleAddFriendForm } from "./redux/reducers";
import Map from "./components/Map/Map";
import AddFriendForm from "./components/AddFriendForm/AddFriendForm";
import "./App.css";

import glamorous, { ThemeProvider } from "glamorous";

const theme = {
  primaryLight: "#D8D8F6",
  primaryDark: "#B18FCF"
};

const Header = glamorous.div(({ theme }) => ({
  displayName: "Header",
  width: "100%",
  padding: "0.5rem 1rem",
  background: `${theme.primaryLight}`,
  " h1": {
    fontFamily: "bebas",
    color: `${theme.primaryDark}`
  },
  " h2": {
    fontWeight: "normal"
  }
}));

export const AddFriendButton = glamorous.button(({ theme }) => ({
  displayName: "AddFriendButton",
  width: "100%",
  background: `${theme.primaryDark}`,
  padding: "0.5rem",
  color: "#eee"
}));

export class App extends React.Component {
  render() {
    const {
      user: { name, friends },
      toggleAddFriendForm,
      showAddFriendForm
    } = this.props;
    return (
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Header>
            <h1> Welcome {name && name}!</h1>
            <h2>Lets find your friends!</h2>
          </Header>
          <Map friends={friends} />
          {!showAddFriendForm && (
            <AddFriendButton className="button" onClick={toggleAddFriendForm}>
              Add a Friend
            </AddFriendButton>
          )}
          <AddFriendForm isOpened={this.state && showAddFriendForm} />
        </React.Fragment>
      </ThemeProvider>
    );
  }
}

export default connect(
  state => {
    console.log(state);
    return state;
  },
  { toggleAddFriendForm }
)(App);
