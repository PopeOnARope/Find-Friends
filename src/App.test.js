import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { App } from "./App";

configure({ adapter: new Adapter() });

const mockProps = {
  user: {
    name: "John Pope",
    friends: [
      {
        avatar: "https://placekitten.com/g/35/35",
        name: "Jim",
        info:
          "Lorem ipsum dolor sit amet, cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
        lat: 32.796421,
        lng: -79.9312164
      },
      {
        avatar: "https://placekitten.com/g/35/36",
        name: "James",
        info:
          "Lorem ipsum dolor sit amet, cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
        lat: 32.795421,
        lng: -79.9412164
      },
      {
        avatar: "https://placekitten.com/g/35/34",
        name: "Jenny",
        info:
          "Lorem ipsum dolor sit amet, cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
        lat: 32.794421,
        lng: -79.9612164
      }
    ]
  },
  showAddFriendForm: false,
  themes: {
    blue: {
      primaryLight: "#d4d8d4",
      primaryDark: "#326ada",
      fontFamilyH1: "bebas"
    },
    CofC: {
      primaryDark: "#660000",
      primaryLight: "#eee",
      fontFamilyH1: "Lobster"
    },
    mcDonalds: {
      primaryDark: "#E84855",
      primaryLight: "#FFFD82",
      fontFamilyH1: "roboto"
    },
    green: {
      primaryDark: "#28965A",
      primaryLight: "#7CFEF0",
      fontFamilyH1: "Georgia"
    }
  },
  theme: "blue"
};

it("renders without crashing", () => {
  const wrapper = shallow(<App {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("shows the add friend form", () => {
  const wrapper = shallow(<App {...mockProps} showAddFriendForm={true} />);
  expect(wrapper).toMatchSnapshot();
});

it("toggles the new friend form when the add friend button is clicked", () => {
  const toggleAddFriendForm = jest.fn();
  const wrapper = shallow(
    <App {...mockProps} toggleAddFriendForm={toggleAddFriendForm} />
  );
  wrapper.find(".button").simulate("click");
  expect(toggleAddFriendForm).toHaveBeenCalled();
  expect(wrapper).toMatchSnapshot();
});
