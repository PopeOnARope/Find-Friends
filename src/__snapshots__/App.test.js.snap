// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    showAddFriendForm={false}
    user={
        Object {
            "friends": Array [
              Object {
                "avatar": "https://placekitten.com/g/25/26",
                "lat": 32.796421,
                "lng": -79.9312164,
                "name": "Jim",
              },
            ],
            "name": "John Pope",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown>
        <glamorous(div)>
                <h1>
                         Welcome 
                        John Pope
                        !
                </h1>
                <h2>
                        Lets find your friends!
                </h2>
        </glamorous(div)>
        <withProps(withScriptjs(withGoogleMap(Map)))
                friends={
                        Array [
                                Object {
                                  "avatar": "https://placekitten.com/g/25/26",
                                  "lat": 32.796421,
                                  "lng": -79.9312164,
                                  "name": "Jim",
                                },
                              ]
                }
        />
        <glamorous(button)
                className="button"
                onClick={undefined}
        >
                Add a Friend
        </glamorous(button)>
        <Connect(AddFriendForm)
                isOpened={null}
        />
</Unknown>,
      "theme": Object {
        "primaryDark": "#B18FCF",
        "primaryLight": "#D8D8F6",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <glamorous(div)>
            <h1>
                         Welcome 
                        John Pope
                        !
            </h1>
            <h2>
                        Lets find your friends!
            </h2>
</glamorous(div)>,
          <withProps(withScriptjs(withGoogleMap(Map)))
            friends={
                        Array [
                                    Object {
                                      "avatar": "https://placekitten.com/g/25/26",
                                      "lat": 32.796421,
                                      "lng": -79.9312164,
                                      "name": "Jim",
                                    },
                                  ]
            }
/>,
          <glamorous(button)
            className="button"
            onClick={undefined}
>
            Add a Friend
</glamorous(button)>,
          <Connect(AddFriendForm)
            isOpened={null}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h1>
                 Welcome 
                John Pope
                !
</h1>,
              <h2>
                Lets find your friends!
</h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Welcome ",
                  "John Pope",
                  "!",
                ],
              },
              "ref": null,
              "rendered": Array [
                " Welcome ",
                "John Pope",
                "!",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lets find your friends!",
              },
              "ref": null,
              "rendered": "Lets find your friends!",
              "type": "h2",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "friends": Array [
              Object {
                "avatar": "https://placekitten.com/g/25/26",
                "lat": 32.796421,
                "lng": -79.9312164,
                "name": "Jim",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add a Friend",
            "className": "button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Add a Friend",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpened": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown>
          <glamorous(div)>
                    <h1>
                               Welcome 
                              John Pope
                              !
                    </h1>
                    <h2>
                              Lets find your friends!
                    </h2>
          </glamorous(div)>
          <withProps(withScriptjs(withGoogleMap(Map)))
                    friends={
                              Array [
                                        Object {
                                          "avatar": "https://placekitten.com/g/25/26",
                                          "lat": 32.796421,
                                          "lng": -79.9312164,
                                          "name": "Jim",
                                        },
                                      ]
                    }
          />
          <glamorous(button)
                    className="button"
                    onClick={undefined}
          >
                    Add a Friend
          </glamorous(button)>
          <Connect(AddFriendForm)
                    isOpened={null}
          />
</Unknown>,
        "theme": Object {
          "primaryDark": "#B18FCF",
          "primaryLight": "#D8D8F6",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <glamorous(div)>
              <h1>
                             Welcome 
                            John Pope
                            !
              </h1>
              <h2>
                            Lets find your friends!
              </h2>
</glamorous(div)>,
            <withProps(withScriptjs(withGoogleMap(Map)))
              friends={
                            Array [
                                          Object {
                                            "avatar": "https://placekitten.com/g/25/26",
                                            "lat": 32.796421,
                                            "lng": -79.9312164,
                                            "name": "Jim",
                                          },
                                        ]
              }
/>,
            <glamorous(button)
              className="button"
              onClick={undefined}
>
              Add a Friend
</glamorous(button)>,
            <Connect(AddFriendForm)
              isOpened={null}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h1>
                   Welcome 
                  John Pope
                  !
</h1>,
                <h2>
                  Lets find your friends!
</h2>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " Welcome ",
                    "John Pope",
                    "!",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " Welcome ",
                  "John Pope",
                  "!",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lets find your friends!",
                },
                "ref": null,
                "rendered": "Lets find your friends!",
                "type": "h2",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "friends": Array [
                Object {
                  "avatar": "https://placekitten.com/g/25/26",
                  "lat": 32.796421,
                  "lng": -79.9312164,
                  "name": "Jim",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add a Friend",
              "className": "button",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Add a Friend",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isOpened": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`shows the add friend form 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    showAddFriendForm={true}
    user={
        Object {
            "friends": Array [
              Object {
                "avatar": "https://placekitten.com/g/25/26",
                "lat": 32.796421,
                "lng": -79.9312164,
                "name": "Jim",
              },
            ],
            "name": "John Pope",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown>
        <glamorous(div)>
                <h1>
                         Welcome 
                        John Pope
                        !
                </h1>
                <h2>
                        Lets find your friends!
                </h2>
        </glamorous(div)>
        <withProps(withScriptjs(withGoogleMap(Map)))
                friends={
                        Array [
                                Object {
                                  "avatar": "https://placekitten.com/g/25/26",
                                  "lat": 32.796421,
                                  "lng": -79.9312164,
                                  "name": "Jim",
                                },
                              ]
                }
        />
        <Connect(AddFriendForm)
                isOpened={null}
        />
</Unknown>,
      "theme": Object {
        "primaryDark": "#B18FCF",
        "primaryLight": "#D8D8F6",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <glamorous(div)>
            <h1>
                         Welcome 
                        John Pope
                        !
            </h1>
            <h2>
                        Lets find your friends!
            </h2>
</glamorous(div)>,
          <withProps(withScriptjs(withGoogleMap(Map)))
            friends={
                        Array [
                                    Object {
                                      "avatar": "https://placekitten.com/g/25/26",
                                      "lat": 32.796421,
                                      "lng": -79.9312164,
                                      "name": "Jim",
                                    },
                                  ]
            }
/>,
          false,
          <Connect(AddFriendForm)
            isOpened={null}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h1>
                 Welcome 
                John Pope
                !
</h1>,
              <h2>
                Lets find your friends!
</h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Welcome ",
                  "John Pope",
                  "!",
                ],
              },
              "ref": null,
              "rendered": Array [
                " Welcome ",
                "John Pope",
                "!",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lets find your friends!",
              },
              "ref": null,
              "rendered": "Lets find your friends!",
              "type": "h2",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "friends": Array [
              Object {
                "avatar": "https://placekitten.com/g/25/26",
                "lat": 32.796421,
                "lng": -79.9312164,
                "name": "Jim",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpened": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown>
          <glamorous(div)>
                    <h1>
                               Welcome 
                              John Pope
                              !
                    </h1>
                    <h2>
                              Lets find your friends!
                    </h2>
          </glamorous(div)>
          <withProps(withScriptjs(withGoogleMap(Map)))
                    friends={
                              Array [
                                        Object {
                                          "avatar": "https://placekitten.com/g/25/26",
                                          "lat": 32.796421,
                                          "lng": -79.9312164,
                                          "name": "Jim",
                                        },
                                      ]
                    }
          />
          <Connect(AddFriendForm)
                    isOpened={null}
          />
</Unknown>,
        "theme": Object {
          "primaryDark": "#B18FCF",
          "primaryLight": "#D8D8F6",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <glamorous(div)>
              <h1>
                             Welcome 
                            John Pope
                            !
              </h1>
              <h2>
                            Lets find your friends!
              </h2>
</glamorous(div)>,
            <withProps(withScriptjs(withGoogleMap(Map)))
              friends={
                            Array [
                                          Object {
                                            "avatar": "https://placekitten.com/g/25/26",
                                            "lat": 32.796421,
                                            "lng": -79.9312164,
                                            "name": "Jim",
                                          },
                                        ]
              }
/>,
            false,
            <Connect(AddFriendForm)
              isOpened={null}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h1>
                   Welcome 
                  John Pope
                  !
</h1>,
                <h2>
                  Lets find your friends!
</h2>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " Welcome ",
                    "John Pope",
                    "!",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " Welcome ",
                  "John Pope",
                  "!",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lets find your friends!",
                },
                "ref": null,
                "rendered": "Lets find your friends!",
                "type": "h2",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "friends": Array [
                Object {
                  "avatar": "https://placekitten.com/g/25/26",
                  "lat": 32.796421,
                  "lng": -79.9312164,
                  "name": "Jim",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isOpened": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`toggles the new friend form when the add friend button is clicked 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    showAddFriendForm={false}
    toggleAddFriendForm={[Function]}
    user={
        Object {
            "friends": Array [
              Object {
                "avatar": "https://placekitten.com/g/25/26",
                "lat": 32.796421,
                "lng": -79.9312164,
                "name": "Jim",
              },
            ],
            "name": "John Pope",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown>
        <glamorous(div)>
                <h1>
                         Welcome 
                        John Pope
                        !
                </h1>
                <h2>
                        Lets find your friends!
                </h2>
        </glamorous(div)>
        <withProps(withScriptjs(withGoogleMap(Map)))
                friends={
                        Array [
                                Object {
                                  "avatar": "https://placekitten.com/g/25/26",
                                  "lat": 32.796421,
                                  "lng": -79.9312164,
                                  "name": "Jim",
                                },
                              ]
                }
        />
        <glamorous(button)
                className="button"
                onClick={[Function]}
        >
                Add a Friend
        </glamorous(button)>
        <Connect(AddFriendForm)
                isOpened={null}
        />
</Unknown>,
      "theme": Object {
        "primaryDark": "#B18FCF",
        "primaryLight": "#D8D8F6",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <glamorous(div)>
            <h1>
                         Welcome 
                        John Pope
                        !
            </h1>
            <h2>
                        Lets find your friends!
            </h2>
</glamorous(div)>,
          <withProps(withScriptjs(withGoogleMap(Map)))
            friends={
                        Array [
                                    Object {
                                      "avatar": "https://placekitten.com/g/25/26",
                                      "lat": 32.796421,
                                      "lng": -79.9312164,
                                      "name": "Jim",
                                    },
                                  ]
            }
/>,
          <glamorous(button)
            className="button"
            onClick={[Function]}
>
            Add a Friend
</glamorous(button)>,
          <Connect(AddFriendForm)
            isOpened={null}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h1>
                 Welcome 
                John Pope
                !
</h1>,
              <h2>
                Lets find your friends!
</h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Welcome ",
                  "John Pope",
                  "!",
                ],
              },
              "ref": null,
              "rendered": Array [
                " Welcome ",
                "John Pope",
                "!",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lets find your friends!",
              },
              "ref": null,
              "rendered": "Lets find your friends!",
              "type": "h2",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "friends": Array [
              Object {
                "avatar": "https://placekitten.com/g/25/26",
                "lat": 32.796421,
                "lng": -79.9312164,
                "name": "Jim",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add a Friend",
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add a Friend",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpened": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown>
          <glamorous(div)>
                    <h1>
                               Welcome 
                              John Pope
                              !
                    </h1>
                    <h2>
                              Lets find your friends!
                    </h2>
          </glamorous(div)>
          <withProps(withScriptjs(withGoogleMap(Map)))
                    friends={
                              Array [
                                        Object {
                                          "avatar": "https://placekitten.com/g/25/26",
                                          "lat": 32.796421,
                                          "lng": -79.9312164,
                                          "name": "Jim",
                                        },
                                      ]
                    }
          />
          <glamorous(button)
                    className="button"
                    onClick={[Function]}
          >
                    Add a Friend
          </glamorous(button)>
          <Connect(AddFriendForm)
                    isOpened={null}
          />
</Unknown>,
        "theme": Object {
          "primaryDark": "#B18FCF",
          "primaryLight": "#D8D8F6",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <glamorous(div)>
              <h1>
                             Welcome 
                            John Pope
                            !
              </h1>
              <h2>
                            Lets find your friends!
              </h2>
</glamorous(div)>,
            <withProps(withScriptjs(withGoogleMap(Map)))
              friends={
                            Array [
                                          Object {
                                            "avatar": "https://placekitten.com/g/25/26",
                                            "lat": 32.796421,
                                            "lng": -79.9312164,
                                            "name": "Jim",
                                          },
                                        ]
              }
/>,
            <glamorous(button)
              className="button"
              onClick={[Function]}
>
              Add a Friend
</glamorous(button)>,
            <Connect(AddFriendForm)
              isOpened={null}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h1>
                   Welcome 
                  John Pope
                  !
</h1>,
                <h2>
                  Lets find your friends!
</h2>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " Welcome ",
                    "John Pope",
                    "!",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " Welcome ",
                  "John Pope",
                  "!",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lets find your friends!",
                },
                "ref": null,
                "rendered": "Lets find your friends!",
                "type": "h2",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "friends": Array [
                Object {
                  "avatar": "https://placekitten.com/g/25/26",
                  "lat": 32.796421,
                  "lng": -79.9312164,
                  "name": "Jim",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add a Friend",
              "className": "button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add a Friend",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isOpened": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
